src/cascadia/TerminalApp/App.cpp

```diff
@@ -118,6 +111,11 @@ namespace winrt::TerminalApp::implementation
         _tabContent.SizeChanged({ this, &App::_OnContentSizeChanged });
     }
 
+    App::~App()
+    {
+        TraceLoggingUnregister(g_hTerminalAppProvider);

```

Comment:
Can't this technically cause a problem because `g_hTerminalAppProvider` is only set up during `Create()` and not on construction?

Should this check null before trying to unregister it?